server/src/main/java/org/opensearch/common/util/ValidationUtils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+/**
+ * Utils to do validations
+ *
+ * @opensearch.internal
+ */
+public final class ValidationUtils {
+    /**
+     * Asserts that the given object is non-null and returns it.
+     *
+     * @param reference Object to assert on
+     * @return Object if non-null
+     * @throws IllegalArgumentException If object was null
+     */
+    public static <T> T validateNotNull(T reference) {
```

Comment:
I think this is identical to the build-in `Objects#requireNonNull`. Can we used that instead?